### PR TITLE
Fix segfault in interpolation with anisotropic ref ratio

### PR DESCRIPTION
**Pull request by prkkumar**
This PR fixes segfault at two interpolation routines that use anisotropic refinement ratio.

### CODE REVIEW

File: Source/ablastr/coarsen/average.cpp [hunk 1]
```diff
@@ -105,7 +105,13 @@ namespace ablastr::coarsen::average
...
 
         // Number of guard cells to fill on coarse patch and number of components
-        const amrex::IntVect ngrow = (mf_src.nGrowVect() + 1) / crse_ratio;
+    amrex::IntVect ngrow;
+
+        for ( int l = 0; l < AMREX_SPACEDIM; ++l ) {
+            if ( crse_ratio[l] == 1 ) ngrow[l] = mf_src.nGrowVect()[l]; // no coarsening
+            else              ngrow[l] = (mf_src.nGrowVect()[l] + 1) / crse_ratio[l];
+        }
+
```
Comment (EZoni) [inline Source/ablastr/coarsen/average.cpp]: Thank you, @prkkumar! Would this work as well?
```suggestion
        amrex::IntVect ngrow = mf_src.nGrowVect();
        for (int l=0; l < AMREX_SPACEDIM; ++l) {
             if (crse_ratio[l] > 1)  ngrow[l] = (ngrow[l] + 1) / crse_ratio[l];
        }

```

Comment (prkkumar) [reply Source/ablastr/coarsen/average.cpp]: Yes, thank you!

**Review by EZoni — APPROVED** [review]
Thank you, @prkkumar! These fixes look good to me.

File: Source/ablastr/coarsen/average.cpp [hunk 2]
```diff
@@ -105,7 +105,11 @@ namespace ablastr::coarsen::average
                                            "source MultiFab and destination MultiFab have different IndexType");
 
         // Number of guard cells to fill on coarse patch and number of components
-        const amrex::IntVect ngrow = (mf_src.nGrowVect() + 1) / crse_ratio;
+        amrex::IntVect ngrow = mf_src.nGrowVect();
+        for (int l=0; l < AMREX_SPACEDIM; ++l) {
+             if (crse_ratio[l] > 1)  ngrow[l] = (ngrow[l] + 1) / crse_ratio[l];
+        }
+
```
Comment (WeiqunZhang) [inline Source/ablastr/coarsen/average.cpp]:
```suggestion
        const amrex::IntVect ngrow = (mf_src.nGrowVect() + crse_ratio-1) / crse_ratio; // round up int division
```

Comment (WeiqunZhang) [reply Source/ablastr/coarsen/average.cpp]: This also works if crse_ratio is 4 or any integer.

Comment (prkkumar) [reply Source/ablastr/coarsen/average.cpp]: Thanks, @WeiqunZhang !

**Review by EZoni — APPROVED** [review]

**Comment by EZoni** [comment]
Will need to be rebased after #3621 fixes the CI issue.

**Review by WeiqunZhang — APPROVED** [review]

**Pull request merged**